### PR TITLE
feat: support namespace for trial template

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

Add a namespace field for the template.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/827)

### CODE REVIEW

File: pkg/ui/v1alpha3/frontend/src/components/HP/Create/HPParameters.jsx [hunk 1]
```diff
@@ -128,7 +130,7 @@ const HPParameters = (props) => {
             goTemplate: {
                 templateSpec: {
                     configMapName: "trial-template",
-                    configMapNamespace: data.metadata.namespace,
+                    configMapNamespace: deploymentNamespace,
```
Comment (johnugeorge) [inline pkg/ui/v1alpha3/frontend/src/components/HP/Create/HPParameters.jsx]: Is this correct? Can't the template be outside kubeflow namespace?

Comment (gaocegege) [reply pkg/ui/v1alpha3/frontend/src/components/HP/Create/HPParameters.jsx]: I think the config map should be kept in kubeflow namespace. We have some built-in templates. If we use data.metadata.namespace, we cannot get them if users use other namespaces.

**Comment by gaocegege** [comment]
Need to add an `onChange` to fetch trial templates.

File: pkg/util/v1alpha3/katibclient/katib_client.go [hunk 2]
```diff
@@ -150,7 +151,10 @@ func (k *KatibClient) GetTrialTemplates(namespace ...string) (map[string]string,
 	ns := getNamespace(namespace...)
 	trialTemplates := &apiv1.ConfigMap{}
 
-	if err := k.client.Get(context.Background(), types.NamespacedName{Name: experimentsv1alpha3.DefaultTrialConfigMapName, Namespace: ns}, trialTemplates); err != nil {
+	err := k.client.Get(context.TODO(), types.NamespacedName{Name: experimentsv1alpha3.DefaultTrialConfigMapName, Namespace: ns}, trialTemplates)
```
Comment (hougangliu) [inline pkg/util/v1alpha3/katibclient/katib_client.go]: `return k.GetConfigMap(experimentsv1alpha3.DefaultTrialConfigMapName, namespace)`

BTW, now trialTemplate can be stored in any configmap, should we always get/update it in default ConfigMap?

**Comment by gaocegege** [comment]
Peek 2019-09-28 09-49

File: pkg/util/v1alpha3/katibclient/katib_client.go (hunk 2 above)
Comment (gaocegege) [reply pkg/util/v1alpha3/katibclient/katib_client.go]: I can refactor it in another PR. Now #774 is in higher priority.

**Comment by gaocegege** [comment]
/cc @hougangliu 

Addressed the comments, PTAL

File: pkg/util/v1alpha3/katibclient/katib_client.go (hunk 2 above)
Comment (hougangliu) [reply pkg/util/v1alpha3/katibclient/katib_client.go]: sure

**Pull request merged**